src/Modules/Finance/Header/Header.tsx

```diff
@@ -26,24 +25,29 @@ export class Header extends React.Component<IHeaderProps, IHeaderState> {
         this.setState({targetSumConfirmed: true});
     };
 
-    targetSumInputChangedHandler = (event: any) => {
-        this.setState({targetSumValue: event.target.value});
+    targetSumInputChangedHandler = (e: React.FormEvent<EventTarget>): void => {
+        const target = e.target as HTMLInputElement;
+        this.setState({targetSumValue: target.value});
     };
 
     render() {
+        const {
+            targetSumValue,
+            targetSumConfirmed,
+            currentTransactionValue,
+            totalSum
+        } = this.state;
         return (
             <div className={styles.Header}>
                 <TargetSum
-                    value={this.state.targetSumValue}
-                    buttonValue={
-                        this.state.targetSumConfirmed ? 'Correct' : 'Confirm'
-                    }
+                    value={targetSumValue}
+                    buttonValue={targetSumConfirmed ? 'Correct' : 'Confirm'}
                     clicked={this.targetSumConfirmedHandler}
                     changed={this.targetSumInputChangedHandler}
-                    confirmed={this.state.targetSumConfirmed}

```

Comment:
Just for me to understand: should I comment things like unneccessary 'this.state' in this line or it's too obvious to be commented?

Reply:
From my point you should comment as much as you can. But plz explane more broader what is wrong with this piece of code?